modulus/utils/graphcast/graph.py

```diff
@@ -132,53 +118,31 @@ def create_g2m_graph(self, verbose: bool = True) -> Tensor:
             Graph2mesh graph.
         """
         # get the max edge length of icosphere with max order
-        edge_src = self.icospheres["order_" + str(self.max_order) + "_vertices"][
-            self.icospheres["order_" + str(self.max_order) + "_faces"][:, 0]
-        ]
-        edge_dst = self.icospheres["order_" + str(self.max_order) + "_vertices"][
-            self.icospheres["order_" + str(self.max_order) + "_faces"][:, 1]
-        ]
-        edge_len_1 = np.max(get_edge_len(edge_src, edge_dst))
-        edge_src = self.icospheres["order_" + str(self.max_order) + "_vertices"][
-            self.icospheres["order_" + str(self.max_order) + "_faces"][:, 0]
-        ]
-        edge_dst = self.icospheres["order_" + str(self.max_order) + "_vertices"][
-            self.icospheres["order_" + str(self.max_order) + "_faces"][:, 2]
-        ]
-        edge_len_2 = np.max(get_edge_len(edge_src, edge_dst))
-        edge_src = self.icospheres["order_" + str(self.max_order) + "_vertices"][
-            self.icospheres["order_" + str(self.max_order) + "_faces"][:, 1]
-        ]
-        edge_dst = self.icospheres["order_" + str(self.max_order) + "_vertices"][
-            self.icospheres["order_" + str(self.max_order) + "_faces"][:, 2]
-        ]
-        edge_len_3 = np.max(get_edge_len(edge_src, edge_dst))
-        edge_len = max([edge_len_1, edge_len_2, edge_len_3])
+
+        max_edge_len = max_edge_length(
+            self.finest_mesh_vertices, self.finest_mesh_src, self.finest_mesh_dst
+        )
 
         # create the grid2mesh bipartite graph
```

Comment:
since we are revisiting this: we might also explore to generalize the grid2mesh / mesh2grid routines a bit
AIFS e.g. uses the same definition in both "mapping" layers leading to the same number of edges in both graphs while GraphCast for some reason uses a non-symmetrical approach
we basically should have
- nearest neighbor grid-> mesh (and optional filtering based on max_edge_len)
- centroid based
- and then it's up to the configuration to specify these options
Would this make sense?

Reply:
I have doubts about this. GraphCast's approach for constructing grid2mesh and mesh2grid edges are significantly different, leading to 3x more edges in mesh2grid compared to grid2mesh. Although I agree that the underlying reason for such design is unclear. Since our immediate goal is to compare directly with GraphCast, I suggest we stick to their design, and consider/revisit this in future enhancements post comparison. Would yo agree?

Reply:
Fully agree with you regarding GraphCast and how they use things. My point was rather potentially future proofing things. For now, GraphCast definitely will just use the mappings as defined in the original version for sure. 
But good point, let's revisit things later.